### PR TITLE
Refactor `GetCluster` function

### PR DESCRIPTION
**Pull request by oriAdler**
Related: [SDA-6551](https://issues.redhat.com/browse/SDA-6551)

### CODE REVIEW

File: vendor/modules.txt [hunk 1]
```diff
@@ -98,6 +98,7 @@ github.com/golang/protobuf/ptypes/any
 github.com/golang/protobuf/ptypes/duration
 github.com/golang/protobuf/ptypes/timestamp
 # github.com/google/uuid v1.3.0
+## explicit
```
Comment (tzvatot) [inline vendor/modules.txt:101]: What do you mean by this comment?

Comment (oriAdler) [reply vendor/modules.txt:101]: TBH not sure, it was added automatically, let me check.

Comment (oriAdler) [reply vendor/modules.txt:101]: This is a related thread https://github.com/golang/go/issues/37946#issuecomment-601210447
I didn't understand the `explicit` comment completely, but I can see we already have a bunch of these comments in this file.

**Pull request merged**